lib/improver/wind_downscaling.py

```diff
@@ -0,0 +1,967 @@
+        if (
+                (a_cube.coord(x_name).units != exp_unit) or
+                (a_cube.coord(y_name).units != exp_unit)):
+            raise ValueError("cube axis have units different from m.")
+        return (xres + yres) / 2.0
+
+    @staticmethod
+    def check_ancils(a_over_s_cube, sigma_cube, z0_cube, pp_oro_cube,
+                     model_oro_cube):
+        """Check ancils grid and units.
+
+        Check if ancil cubes are on the same grid and if they have the
+        expected units. The testing for "same grid" might be replaced
+        if there is a general utils function made for it or so.
+
+        Parameters:
+        ----------
+        a_over_s_cube: iris cube
+            holding the silhoutte roughness field
+        sigma_cube: iris cube
+            holding the standard deviation of height in a grid cell
+        z0_cube: iris cube or None
+            holding the vegetative roughness field
+        pp_oro_cube: iris cube
+            holding the post processing grid orography
+        model_oro_cube: iris cube
+            holding the model orography on post processing grid
+
+        Returns:
+        -------
+        logical
+            describing whether or not the tests passed
+
+        """
+        alist = [a_over_s_cube, sigma_cube, pp_oro_cube, model_oro_cube]
+        unwanted_coord_list = [
+            "time", "height", "model_level_number", "forecast_time",
+            "forecast_reference_time", "forecast_period"]
+        for field, exp_unit in zip(alist, [None, Unit("m"), Unit("m"),
+                                           Unit("m")]):
+            for unwanted_coord in unwanted_coord_list:
+                try:
+                    field.remove_coord(unwanted_coord)
+                except Exception:
+                    pass
+            if field.units != exp_unit:
+                msg = ('{} ancil field has unexpected unit:'
+                       ' {} (expected) vs. {} (actual)')
+                raise ValueError(
+                    msg.format(field.name(), exp_unit, field.units))
+        if z0_cube is not None:
+            alist.append(z0_cube)
+            for unwanted_coord in unwanted_coord_list:
+                try:
+                    z0_cube.remove_coord(unwanted_coord)
+                except Exception:
+                    pass
+            if z0_cube.units != Unit('m'):
+                msg = ("z0 ancil has unexpected unit: should be {} "
+                       "is {}")
+                raise ValueError(msg.format(Unit('m'), z0_cube.units))
+        mlist = list(itertools.permutations(alist, 2))
+        oklist = []
+        for entr in mlist:
+            oklist.append(entr[0].coords() == entr[1].coords())
+            # HybridHeightToPhenomOnPressure._cube_compatibility_check(entr[0],
+            # entr[1])
+        return np.array(oklist).all()  # replace by a return value of True
+
+    def find_coord_order(self, mcube):
```

Comment:
I think the loop for looping over the coordinates could potentially be removed using the Iris [coord_dims](http://scitools.org.uk/iris/docs/latest/iris/iris/cube.html#iris.cube.Cube.coord_dims) functionality, which may be preferable i.e. 
```python
xpos, = mcube.coord_dims(self.x_name)
ypos, = mcube.coord_dims(self.y_name)
zpos, = mcube.coord_dims(self.z_name)
tpos, = mcube.coord_dims(self.t_name)
```